### PR TITLE
Reapply [libcxx] [modules] Fix relative paths with absolute LIBCXX_INSTALL_MODULES_DIR

### PR DESCRIPTION
**Pull request by mstorsjo**
This reapplies 272d1b44efdedb68c194970a610f0ca1b7b769c5 (from #85756), which was reverted in
407937036fa7640f61f225474b1ea6623a40dbdd.

In the previous attempt, empty CMAKE_INSTALL_PREFIX was handled by quoting them, in d209d1340b99d4fbd325dffb5e13b757ab8264ea. That made the calls to cmake_path(ABSOLUTE_PATH) succeed, but the output paths of that weren't actually absolute, which was required by file(RELATIVE_PATH).

Avoid this issue by appending a slash to CMAKE_INSTALL_PREFIX, so we always get an absolute path, even if CMAKE_INSTALL_PREFIX was empty.

### CODE REVIEW

**Review by ilovepi — APPROVED** [review]
LGTM. This seems like a sound approach to me but you may want to double check with Petr.

File: libcxx/modules/CMakeLists.txt
Comment (mordante) [inline libcxx/modules/CMakeLists.txt:213]: I wonder what happens when `${CMAKE_INSTALL_PREFIX}` is "/", then we get "//". I'm not sure whether that is a valid path. How about something like
```
if(${CMAKE_INSTALL_PREFIX} STREQUAL "")
  set(BASE_DIRECTORY "/" PARENT_SCOPE)
else()
  set(BASE_DIRECTORY ${CMAKE_INSTALL_PREFIX} PARENT_SCOPE)
endif()
```

Comment (mstorsjo) [reply libcxx/modules/CMakeLists.txt:213]: The double slash path here is effectively harmless; after `cmake_path(ABSOLUTE_PATH)`, the output only has a single leading slash even in that case.

But you're right that it's probably nicer to be explicit about this.

**Review by mordante — APPROVED** [review]
Thanks LGTM! Please wait for @petrhosek's approval before landing.

**Comment by mstorsjo** [comment]
> Thanks LGTM! Please wait for @petrhosek's approval before landing.

Sure, although I don't know if we need his extra input here any longer :-)

**Comment by mordante** [comment]
> > Thanks LGTM! Please wait for @petrhosek's approval before landing.
> 
> Sure, although I don't know if we need his extra input here any longer :-)

I thought Petr had reported the issue with the previous version. It seems I was mistaken it was @ilovepi. Since @ilovepi has approved I'm happy to land it.

**Comment by ilovepi** [comment]
Yeah I think we’re good here. We definitely use an empty install directory so the fix makes sense to me. I’ll keep an eye on our ci, but I think our buildout should catch this too.

**Comment by mstorsjo** [comment]
FWIW, the fuchsia build seems to stay green now, so I'll request a backport of this again.

/cherry-pick 50801f1095d33e712c3a51fdeef82569bd09007f

**Pull request merged**
